R/adverb-partial.R

```diff
@@ -98,11 +98,12 @@ partial <- function(.f,
   fn_expr <- enexpr(.f)
   .fn <- switch(typeof(.f),
     builtin = ,
-    special =
-      as_closure(.f),
-    closure =
-      .f,
-    abort(sprintf("`.f` must be a function, not %s", friendly_type_of(.f)))
+    special = as_closure(.f),
+    closure = .f,
+    {
+      f <- .f
+      cli::cli_abort("{.arg .f} must be a function, not {.obj_friendly_type {f}}")
```

Comment:
There is a typo in the class name. Should pass `arg = `.

Probably best to use `check_function()` from the rlang compat file instead.